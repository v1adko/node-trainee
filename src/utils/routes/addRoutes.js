const addRoutes = router => (routes) => {
  routes.forEach(route => route(router));
  return router;
};

export default addRoutes;
