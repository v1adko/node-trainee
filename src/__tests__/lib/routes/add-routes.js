import addRoutes from '../../../lib/routes//add-routes';

function testController() {}

function controllerMethod() {}

const routeConfig = [
  {
    method: 'get',
    route: '/firstTestRoute',
    testController,
    controllerMethod
  },
  {
    method: 'post',
    route: '/secondTestRoute',
    testController,
    controllerMethod
  }
];

describe('Test getRequestData', () => {
  it('should destruct and return all data of request', () => {
    const commonRouter = addRoutes(routeConfig);

    expect(commonRouter).toMatchSnapshot();
  });
});
