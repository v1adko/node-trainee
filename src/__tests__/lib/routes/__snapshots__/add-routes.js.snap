// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test getRequestData should destruct and return all data of request 1`] = `[Function]`;
